src/execution/operator/csv_scanner/encode/csv_encoder.cpp

```diff
@@ -36,10 +37,23 @@ void CSVEncoderBuffer::Reset() {
 	actual_encoded_buffer_size = 0;
 }
 
-CSVEncoder::CSVEncoder(const DBConfig &config, const string &encoding_name_to_find, idx_t buffer_size) {
+CSVEncoder::CSVEncoder(ClientContext &context, const string &encoding_name_to_find, idx_t buffer_size)
+    : pass_on_byte(0) {
+	auto &config = DBConfig::GetConfig(context);
 	encoding_name = StringUtil::Lower(encoding_name_to_find);
 	auto function = config.GetEncodeFunction(encoding_name_to_find);
 	if (!function) {
+		if (!context.db->ExtensionIsLoaded("encodings")) {
+			// Maybe we can try to auto-load from our encodings extension, if this somehow fails, we just error.
+			auto extension_loaded = ExtensionHelper::TryAutoLoadExtension(context, "encodings");
```

Comment:
Could you test whether / why this actually works?

I would have expected to find the `encodings` added to `AUTOLOADABLE_EXTENSIONS`, and ideally also in the list of auto-installable.

I think it's expected to be used guarded like:
```
                        if (ExtensionHelper::CanAutoloadExtension(extension_name)) {
                                if (ExtensionHelper::TryAutoLoadExtension(context, extension_name)) {
                                       function = // TODO ...;
                                } 
                        }
```

It's weird / error prone, it would be handy to clean that up, but ...